### DISCUSSION TITLE
Redundancy in rule for strings?

### DISCUSSION THREAD

**Discussion by jlkiri** [kdl-org/kdl#13] [Q&A] [answered]
Won't this be sufficient?

```diff
- string := '"' ('\\' ["\\] | [^"])* '"'
+ string := '"' ('\\' ["] | [^"])* '"'
```
 
[Nearly](https://omrelli.ug/nearley-playground/) version for testing: `string -> "\"" ("\\" [\"] | [^"]):* "\""`

**Comment by jlkiri** [comment]
Apparently the grammar changed a bit while I was writing this. Should be OK.

**Reply by zkat** [reply]
haha yeah I literally just finished writing the actual parser and updated the grammar in the process. I'm working on tests now and will push it shortly 👍🏻

**Comment by zkat** [comment, marked as answer]
The string grammar has been fixed and expanded so this should be done :)

**Comment by larsgw** [comment]
The reason I put it in like that is to be able to escape backslashes at the end of the string. Otherwise `"foo\\"` would not end the string, though backtracking may circumvent the problem entirely.
